controllers/storageclassclaimsecret/storageclassclaimsecret_controller.go

```diff
@@ -0,0 +1,691 @@
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// StorageClassClaimSecretReconciler reconciles a Secret containing a StorageClassClaim object
+// nolint
+type StorageClassClaimSecretReconciler struct {
+	client.Client
+	Scheme            *runtime.Scheme
+	OperatorNamespace string
+
+	log                          logr.Logger
+	ctx                          context.Context
+	storageConsumer              *v1alpha1.StorageConsumer
+	storageCluster               *v1.StorageCluster
+	storageClassClaim            *v1alpha1.StorageClassClaim
+	storageClassClaimSecret      *corev1.Secret
+	cephBlockPool                *rookCephv1.CephBlockPool
+	cephFilesystemSubVolumeGroup *rookCephv1.CephFilesystemSubVolumeGroup
+	cephClientProvisioner        *rookCephv1.CephClient
+	cephClientNode               *rookCephv1.CephClient
+	cephResourcesByName          map[string]*v1alpha1.CephResourcesSpec
+	storageProfile               *v1.StorageProfile
+}
+
+// +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=ceph.rook.io,resources=cephclients,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=ceph.rook.io,resources=cephfilesystemsubvolumegroups,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=ceph.rook.io,resources=cephblockpools,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclusters,verbs=get;watch;list
+// +kubebuilder:rbac:groups=storage.k8s.io,resources=storageclasses,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=snapshot.storage.k8s.io,resources=volumesnapshotclasses,verbs=get;list;watch;create;delete
+// +kubebuilder:rbac:groups=core,resources=persistentvolumes,verbs=get;list;watch
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete
+
+func (r *StorageClassClaimSecretReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	r.log = ctrllog.FromContext(ctx, "StorageClassClaim Secret", request)
+	r.ctx = ctrllog.IntoContext(ctx, r.log)
```

Comment:
You are already getting the `r.log` from the context. There is no need to call `IntoContext` as well

Reply:
Fair enough. This is an artifact from the initial file copy, so I dunno why it's there, but fixing it nonetheless.